glambie/processing/processing_helpers.py

```diff
@@ -261,3 +276,96 @@ def extend_annual_timeseries_if_outside_trends_period(annual_timeseries: Timeser
+        will contain more datasets than input data catalogue when gaps were found.
+    """
+    new_datasets = []
+    for timeseries in data_catalogue.datasets:
+        if not timeseries.data.is_cumulative_valid():  # if invalid convert to handle the gaps
+            # 1 split timeseries dataframe
+            df_data = timeseries.data.as_dataframe()
+            split_dataframes = slice_timeseries_at_gaps(df_data)
+            # 2 add split timeseries to new_datasets
+            for idx, split_timeseries in enumerate(split_dataframes):
+                timeseries_copy = timeseries.copy()
+                # rename user group name to be unique
+                timeseries_copy.user_group = f"{timeseries_copy.user_group }_{str(idx+1)}"
+                timeseries_copy.data.changes = np.array(split_timeseries["changes"])
+                timeseries_copy.data.errors = np.array(split_timeseries["errors"])
+                timeseries_copy.data.start_dates = np.array(split_timeseries["start_dates"])
+                timeseries_copy.data.end_dates = np.array(split_timeseries["end_dates"])
+                new_datasets.append(timeseries_copy)
+        else:  # or else append original
+            new_datasets.append(timeseries)
+
+    new_data_catalogue = DataCatalogue.from_list(new_datasets, base_path=data_catalogue.base_path)
+    return new_data_catalogue
+
+
+def slice_timeseries_at_gaps(df_timeseries: pd.DataFrame) -> list[pd.DataFrame]:
+    """
+    Splits a dataframe into separate chunks/slices to remove temporal gaps in the timeseries.
+
+    Parameters
+    ----------
+    df_timeseries : pd.DataFrame
+        pandas dataframe containing a timeseries and the columns 'start_dates' and 'end_dates'.
+        If 'end_date' != 'start_date' for any consecutive rows this is considered as a time gap.
+
+    Returns
+    -------
+    list[pd.DataFrame]
+        a list with slices of the original dataframe. All new dataframes within the list are gapless.
+    """
+    split_indices = []
+    split_timeseries_dataframes = []
+    for idx, end_date in enumerate(df_timeseries["end_dates"].iloc[:-1]):
+        # If end_date != start_date for any of the consecutive rows, append to indices where df is split
+        if end_date != df_timeseries["start_dates"].iloc[idx + 1]:
+            split_indices.append(idx)
+    previous_index = 0
+    for split_index in split_indices:
+        split_timeseries_dataframes.append(df_timeseries.iloc[previous_index:split_index + 1].reset_index(drop=True))
+        previous_index = split_index + 1
+    # plus append last / full split in the end
+    split_timeseries_dataframes.append(df_timeseries.iloc[previous_index:].reset_index(drop=True))
+    return split_timeseries_dataframes
+
+
+def set_unneeded_columns_to_nan(data_catalogue: DataCatalogue) -> DataCatalogue:
+    """
+    Sets data columns of TimeseriesData not needed in algorithm to NaN within a DataCatalogue
+    to simplify object manipulating
```

Comment:
```suggestion
    to simplify object manipulation
```

Reply:
ooft you're so good at spotting random typos and errors in the docstrings! done:)